File: go/military.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// MilitaryService — V3 military/defense domain analysis.
type MilitaryService struct {
	t *internal.Transport
}

func (s *MilitaryService) AntiSpoofing(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/military/anti-spoofing/analyze", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MilitaryService) Classification(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/military/classification/analyze", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MilitaryService) CommandChain(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/military/command-chain/analyze", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MilitaryService) CrossDomain(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/military/cross-domain/analyze", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MilitaryService) Opsec(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/military/opsec/analyze", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MilitaryService) Roe(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/military/roe/analyze", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MilitaryService) TacticalAutonomy(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/military/tactical-autonomy/analyze", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MilitaryService) Orchestrate(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/military/orchestrate", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MilitaryService) Status(ctx context.Context) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "GET", "/v3/military/status", nil, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

File: rust/src/resources/military.rs
use std::sync::Arc;

use reqwest::Method;
use serde_json::Value;

use crate::error::Result;
use crate::transport::Transport;

pub struct Military {
    transport: Arc<Transport>,
}

impl Military {
    pub(crate) fn new(transport: Arc<Transport>) -> Self {
        Self { transport }
    }

    pub async fn anti_spoofing(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(
                Method::POST,
                "/v3/military/anti-spoofing/analyze",
                Some(body),
            )
            .await
    }

    pub async fn classification(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(
                Method::POST,
                "/v3/military/classification/analyze",
                Some(body),
            )
            .await
    }

    pub async fn command_chain(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(
                Method::POST,
                "/v3/military/command-chain/analyze",
                Some(body),
            )
            .await
    }

    pub async fn cross_domain(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(
                Method::POST,
                "/v3/military/cross-domain/analyze",
                Some(body),
            )
            .await
    }

    pub async fn opsec(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(Method::POST, "/v3/military/opsec/analyze", Some(body))
            .await
    }

    pub async fn roe(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(Method::POST, "/v3/military/roe/analyze", Some(body))
            .await
    }

    pub async fn tactical_autonomy(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(
                Method::POST,
                "/v3/military/tactical-autonomy/analyze",
                Some(body),
            )
            .await
    }

    pub async fn orchestrate(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(Method::POST, "/v3/military/orchestrate", Some(body))
            .await
    }

    pub async fn status(&self) -> Result<Value> {
        self.transport
            .request(Method::GET, "/v3/military/status", None::<&()>)
            .await
    }
}
